### PR TITLE
Mon 11583 logv2 engine

### PR DESCRIPTION
**Pull request by rem31**

### CODE REVIEW

File: centreon-engine/src/events/sched_info.cc [hunk 1]
```diff
@@ -127,25 +183,43 @@ void display_scheduling_info() {
...
       << "Last scheduled check:               "
       << ctime(&scheduling_info.last_service_check) << "\n";
+  log_v2::process()->info(
+      "Service interleave factor:          {}\nMax service check spread:       "
+      "    {} min\nFirst scheduled check:              {}Last scheduled check: "
+      "              {}\n",
+      scheduling_info.service_interleave_factor,
+      scheduling_info.max_service_check_spread,
+      ctime(&scheduling_info.first_service_check),
+      ctime(&scheduling_info.last_service_check));
```
Comment (bouda1) [inline centreon-engine/src/events/sched_info.cc:193]: All these previous lines contain too many '\n'

Comment (bouda1) [reply centreon-engine/src/events/sched_info.cc:193]: maybe we should call log_v2::process()->... several times, one for each desired line, or maybe it is better to format a single line with all the informations

File: centreon-engine/src/events/sched_info.cc [hunk 2]
```diff
@@ -35,36 +36,54 @@ using namespace com::centreon::engine::logging;
  */
 void display_scheduling_info() {
   // Notice.
-  logger(log_info_message, basic)
+  engine_logger(log_info_message, basic)
```
Comment (bouda1) [inline centreon-engine/src/events/sched_info.cc:39]: Be careful with logs in this file, too many '\n' to reformat

File: centreon-engine/src/downtimes/service_downtime.cc [hunk 3]
```diff
@@ -281,6 +292,12 @@ int service_downtime::subscribe() {
       << "\n"
          " Trigger ID:  "
       << get_triggered_by();
+  log_v2::downtimes()->trace(
+      " Fixed/Flex:  {} Start:       {} \n End:         {} \n Duration:    {}h "
+      "{}m {}s\n Downtime ID: {}\n Trigger ID:  {}",
+      (is_fixed() ? "Fixed\n" : "Flexible\n"), start_time_string,
+      end_time_string, hours, minutes, seconds, get_downtime_id(),
+      get_triggered_by());
```
Comment (bouda1) [inline centreon-engine/src/downtimes/service_downtime.cc:300]: What to do with '\n'?

File: centreon-engine/src/downtimes/service_downtime.cc [hunk 4]
```diff
@@ -259,14 +266,18 @@ int service_downtime::subscribe() {
...
-                              << found->second->get_description();
-  logger(dbg_downtime, basic)
+  engine_logger(dbg_downtime, basic) << "Scheduled Downtime Details:";
+  log_v2::downtimes()->trace("Scheduled Downtime Details:");
+  engine_logger(dbg_downtime, basic) << " Type:        Service Downtime\n"
+                                        " Host:        "
+                                     << found->second->get_hostname()
+                                     << "\n"
+                                        " Service:     "
+                                     << found->second->get_description();
```
Comment (bouda1) [inline centreon-engine/src/downtimes/service_downtime.cc:276]: Issues with '\n'

File: centreon-engine/modules/external_commands/src/processing.cc [hunk 5]
```diff
@@ -639,16 +643,25 @@ bool processing::execute(const std::string& cmdstr) const {
...
 
-  logger(dbg_external_command, more) << "External command id: " << command_id
-                                     << "\nCommand entry time: " << entry_time
-                                     << "\nCommand arguments: " << args;
+  engine_logger(dbg_external_command, more)
+      << "External command id: " << command_id
+      << "\nCommand entry time: " << entry_time
+      << "\nCommand arguments: " << args;
+  log_v2::external_command()->debug(
+      "External command id: {} \nCommand entry time: {} \nCommand arguments: ",
```
Comment (bouda1) [inline centreon-engine/modules/external_commands/src/processing.cc:663]: {} missing

File: centreon-engine/src/downtimes/host_downtime.cc [hunk 6]
```diff
@@ -230,11 +237,14 @@ int host_downtime::subscribe() {
...
-                              << hst->get_name();
-  logger(dbg_downtime, basic)
+  engine_logger(dbg_downtime, basic) << "Scheduled Downtime Details:";
+  log_v2::downtimes()->trace("Scheduled Downtime Details:");
+  engine_logger(dbg_downtime, basic) << " Type:        Host Downtime\n"
+                                        " Host:        "
+                                     << hst->get_name();
+  log_v2::downtimes()->trace(" Type:        Host Downtime\n Host:        {}",
+                             hst->get_name());
+  engine_logger(dbg_downtime, basic)
```
Comment (bouda1) [inline centreon-engine/src/downtimes/host_downtime.cc:247]: What to do with '\n'?

File: centreon-engine/src/downtimes/host_downtime.cc [hunk 7]
```diff
@@ -249,6 +259,11 @@ int host_downtime::subscribe() {
       << "\n"
          " Trigger ID:  "
       << get_triggered_by();
+  log_v2::downtimes()->trace(
+      " Fixed/Flex:  {} Start:       {}\n End:         {}\n Duration:    {}h "
+      "{}m {}s\n Downtime ID: {}\n Trigger ID:  ",
+      (is_fixed() ? "Fixed\n" : "Flexible\n"), start_time_string,
+      end_time_string, hours, minutes, get_downtime_id(), get_triggered_by());
```
Comment (bouda1) [inline centreon-engine/src/downtimes/host_downtime.cc:266]: issue with \n

**Review by bouda1 — CHANGES_REQUESTED** [review]

File: centreon-engine/src/host.cc [hunk 8]
```diff
@@ -2822,14 +3026,22 @@ int host::process_check_result_3x(enum host::host_state new_state,
...
       << get_max_attempts() << ", CHECK TYPE="
       << (get_check_type() == check_active ? "ACTIVE" : "PASSIVE")
       << ", STATE TYPE=" << (get_state_type() == hard ? "HARD" : "SOFT")
       << ", OLD STATE=" << get_current_state() << ", NEW STATE=" << new_state;
+  log_v2::checks()->debug(
+      "HOST: {}, ATTEMPT={}/{}, CHECK TYPE={}, STATE TYPE={}, OLD STATE={}, "
+      "NEW STATE={}",
+      _name, get_current_attempt(), get_max_attempts(),
+      (get_check_type() == check_active ? "ACTIVE" : "PASSIVE"),
+      (get_state_type() == hard ? "HARD" : "SOFT"), get_current_state(),
```
Comment (bouda1) [inline centreon-engine/src/host.cc:3043]: parenthesis

File: centreon-engine/src/host.cc [hunk 9]
```diff
@@ -2872,9 +3087,12 @@ int host::process_check_result_3x(enum host::host_state new_state,
...
 
-      logger(dbg_checks, more) << "Host experienced a "
-                               << (get_state_type() == hard ? "HARD" : "SOFT")
-                               << " recovery (it's now UP).";
+      engine_logger(dbg_checks, more)
+          << "Host experienced a "
+          << (get_state_type() == hard ? "HARD" : "SOFT")
+          << " recovery (it's now UP).";
+      log_v2::checks()->debug("Host experienced a {} recovery (it's now UP).",
+                              (get_state_type() == hard ? "HARD" : "SOFT"));
```
Comment (bouda1) [inline centreon-engine/src/host.cc:3095]: parenthesis

File: centreon-engine/src/host.cc [hunk 10]
```diff
@@ -3209,20 +3469,30 @@ int host::process_check_result_3x(enum host::host_state new_state,
...
+  engine_logger(dbg_checks, more)
+      << "Pre-handle_host_state() Host: " << _name
+      << ", Attempt=" << get_current_attempt() << "/" << get_max_attempts()
+      << ", Type=" << (get_state_type() == hard ? "HARD" : "SOFT")
+      << ", Final State=" << _current_state;
+  log_v2::checks()->debug(
+      "Pre-handle_host_state() Host: {}, Attempt={}/{}, Type={}, Final "
+      "State={}",
+      _name, get_current_attempt(), get_max_attempts(),
+      (get_state_type() == hard ? "HARD" : "SOFT"), _current_state);
```
Comment (bouda1) [inline centreon-engine/src/host.cc:3481]: parenthesis

File: centreon-engine/src/host.cc [hunk 11]
```diff
@@ -3209,20 +3469,30 @@ int host::process_check_result_3x(enum host::host_state new_state,
...
+  engine_logger(dbg_checks, more)
+      << "Post-handle_host_state() Host: " << _name
+      << ", Attempt=" << get_current_attempt() << "/" << get_max_attempts()
+      << ", Type=" << (get_state_type() == hard ? "HARD" : "SOFT")
+      << ", Final State=" << _current_state;
+  log_v2::checks()->debug(
+      "Post-handle_host_state() Host: {}, Attempt={}/{}, Type={}, Final "
+      "State={}",
+      _name, get_current_attempt(), get_max_attempts(),
+      (get_state_type() == hard ? "HARD" : "SOFT"), _current_state);
```
Comment (bouda1) [inline centreon-engine/src/host.cc:3495]: parenthesis

File: centreon-engine/src/macros/process.cc [hunk 12]
```diff
@@ -42,16 +43,21 @@ int process_macros_r(nagios_macros* mac,
...
-  logger(dbg_macros, more) << "**** BEGIN MACRO PROCESSING ***********\n"
-                              "Processing: '"
-                           << input_buffer << "'";
+  engine_logger(dbg_macros, more) << "**** BEGIN MACRO PROCESSING ***********\n"
+                                     "Processing: '"
+                                  << input_buffer << "'";
+  log_v2::macros()->trace(
+      "**** BEGIN MACRO PROCESSING ***********\n"
+      "Processing: '{}'",
+      input_buffer);
```
Comment (bouda1) [inline centreon-engine/src/macros/process.cc:60]: \n

File: centreon-engine/src/macros/process.cc [hunk 13]
```diff
@@ -77,30 +83,43 @@ int process_macros_r(nagios_macros* mac,
...
                                       &clean_options, &free_macro);
 
-          logger(dbg_macros, most)
+          engine_logger(dbg_macros, most)
               << "  Processed '" << token.c_str() << "', To '" << token_resolved
               << "', Clean Options: " << clean_options
               << ", Free: " << free_macro;
+          log_v2::macros()->trace(
+              "  Processed '{}', To '{}', Clean Options: {}, Free: {}",
+              token.c_str(), token_resolved, clean_options, free_macro);
```
Comment (bouda1) [inline centreon-engine/src/macros/process.cc:92]: c_str() not needed

File: centreon-engine/src/macros/process.cc [hunk 14]
```diff
@@ -144,8 +172,12 @@ int process_macros_r(nagios_macros* mac,
...
-  logger(dbg_macros, more) << "  Done.  Final output: '" << output_buffer
-                           << "'\n"
-                              "**** END MACRO PROCESSING *************";
+  engine_logger(dbg_macros, more) << "  Done.  Final output: '" << output_buffer
+                                  << "'\n"
+                                     "**** END MACRO PROCESSING *************";
+  log_v2::macros()->trace(
+      "  Done.  Final output: '{}'\n"
+      "**** END MACRO PROCESSING *************",
+      output_buffer);
```
Comment (bouda1) [inline centreon-engine/src/macros/process.cc:181]: \n

File: centreon-engine/src/main.cc [hunk 15]
```diff
@@ -173,35 +174,65 @@ int main(int argc, char* argv[]) {
...
+          "MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the "
+          "GNU\n"
+          "General Public License for more details.\n"
+          "\n"
+          "You should have received a copy of the GNU General Public "
+          "License\n"
+          "along with this program. If not, see\n"
+          "<http://www.gnu.org/licenses/>.",
+          CENTREON_ENGINE_VERSION_STRING);
+
```
Comment (bouda1) [inline centreon-engine/src/main.cc:230]: std::cout looks better than log_v2 here. The goal of these messages is to display them in the terminal

File: centreon-engine/src/main.cc [hunk 16]
```diff
@@ -229,6 +261,38 @@ int main(int argc, char* argv[]) {
...
+          "  -D, --diagnose              Generate a diagnostic file.\n"
+          "\n"
+          "Online:\n"
+          "  Website                     https://www.centreon.com\n"
+          "  Reference documentation     "
+          "https://documentation.centreon.com/docs/centreon-engine/en/"
+          "latest/\n"
+          "  Sources                     "
+          "https://github.com/centreon/centreon-engine",
+          argv[0]);
```
Comment (bouda1) [inline centreon-engine/src/main.cc:294]: the same here

File: centreon-engine/src/main.cc [hunk 17]
```diff
@@ -269,14 +333,59 @@ int main(int argc, char* argv[]) {
...
+            "Total Errors: {}",
+            commands::command::commands.size(),
+            commands::connector::connectors.size(), contact::contacts.size(),
+            hostdependency::hostdependencies.size(),
+            hostescalation::hostescalations.size(),
+            hostgroup::hostgroups.size(), host::hosts.size(),
+            servicedependency::servicedependencies.size(),
+            serviceescalation::serviceescalations.size(),
+            servicegroup::servicegroups.size(), service::services.size(),
+            timeperiod::timeperiods.size(), config_warnings, config_errors);
```
Comment (bouda1) [inline centreon-engine/src/main.cc:369]: same here I think

File: centreon-engine/src/main.cc [hunk 18]
```diff
@@ -269,14 +333,59 @@ int main(int argc, char* argv[]) {
...
+            "Check your configuration file(s) to ensure that they contain "
+            "valid directives and data definitions. If you are upgrading from "
+            "a previous version of Centreon Engine, you should be aware that "
+            "some variables/definitions may have been removed or modified in "
+            "this version. Make sure to read the documentation regarding the "
+            "config files, as well as the version changelog to find out what "
+            "has changed.");
+        engine_logger(logging::log_config_error, logging::basic)
+            << "One or more problems occurred while processing the config "
+               "files.\n\n" ERROR_CONFIGURATION;
```
Comment (bouda1) [inline centreon-engine/src/main.cc:388]: to be display on stdout or stderr

File: centreon-engine/src/main.cc [hunk 19]
```diff
@@ -296,7 +405,10 @@ int main(int argc, char* argv[]) {
           try {
             p.parse(config.state_retention_file(), state);
           } catch (std::exception const& e) {
-            logger(logging::log_config_error, logging::basic) << e.what();
+            log_v2::config()->error("Error while parsing the retention: {}",
+                                    e.what());
+            engine_logger(logging::log_config_error, logging::basic)
+                << e.what();
```
Comment (bouda1) [inline centreon-engine/src/main.cc:411]: idem here

File: centreon-engine/src/notifier.cc [hunk 20]
```diff
@@ -156,9 +157,11 @@ notifier::notifier(notifier::notifier_type notifier_type,
       _state_history{{}},
       _pending_flex_downtime{0} {
   if (retry_interval <= 0) {
-    logger(log_config_error, basic)
+    engine_logger(log_config_error, basic)
         << "Error: Invalid notification_interval value for notifier '"
         << display_name << "'";
+    log_v2::config()->error(
+        "Error: Invalid notification_interval value for notifier '{}'");
```
Comment (bouda1) [inline centreon-engine/src/notifier.cc:164]: variable missing

File: centreon-engine/src/notifier.cc [hunk 21]
```diff
@@ -262,68 +278,93 @@ bool notifier::_is_notification_viable_normal(reason_type type
...
           now) {
-    logger(dbg_notifications, more)
+    engine_logger(dbg_notifications, more)
         << "This notifier is configured with a first notification delay, we "
            "won't send notification until timestamp "
         << (_first_notification_delay * config->interval_length());
+    log_v2::notifications()->debug(
+        "This notifier is configured with a first notification delay, we "
+        "won't send notification until timestamp {}",
+        (_first_notification_delay * config->interval_length()));
```
Comment (bouda1) [inline centreon-engine/src/notifier.cc:357]: parenthesis

File: centreon-engine/src/notifier.cc [hunk 22]
```diff
@@ -332,20 +373,28 @@ bool notifier::_is_notification_viable_normal(reason_type type
...
             now) {
-          logger(dbg_notifications, more)
+          engine_logger(dbg_notifications, more)
               << "This notifier problem has been sent at " << _last_notification
               << " so it won't be sent until "
               << (notification_interval * config->interval_length());
+          log_v2::notifications()->debug(
+              "This notifier problem has been sent at {} so it won't be sent "
+              "until {}",
+              (notification_interval * config->interval_length()));
```
Comment (bouda1) [inline centreon-engine/src/notifier.cc:397]: missing variable

File: centreon-engine/src/notifier.cc [hunk 23]
```diff
@@ -382,63 +439,91 @@ bool notifier::_is_notification_viable_recovery(reason_type type
...
+      engine_logger(dbg_notifications, more)
           << "This notifier is configured with a recovery notification delay. "
           << "It won't send any recovery notification until timestamp "
           << " so it won't be sent until "
           << (get_last_hard_state_change() + _recovery_notification_delay);
+      log_v2::notifications()->debug(
+          "This notifier is configured with a recovery notification delay. "
+          "It won't send any recovery notification until timestamp "
+          "so it won't be sent until {}",
+          (get_last_hard_state_change() + _recovery_notification_delay));
```
Comment (bouda1) [inline centreon-engine/src/notifier.cc:506]: parenthesis

File: centreon-engine/tests/configuration/applier/applier-log.cc [hunk 24]
```diff
@@ -0,0 +1,492 @@
+/*
+ * Copyright 2019 Centreon (https://www.centreon.com/)
```
Comment (bouda1) [inline centreon-engine/tests/configuration/applier/applier-log.cc:2]: 2021

File: centreon-engine/test/configuration/applier/xrddefault.cc [hunk 25]
```diff
@@ -63,7 +64,7 @@ int xrddefault_grab_config_info(char const* main_config_file) {
 
   /* open the main config file for reading */
   if ((thefile = mmap_fopen(main_config_file)) == NULL) {
-    logger(dbg_retentiondata, most)
+    engine_logger(dbg_retentiondata, most)
```
Comment (bouda1) [inline centreon-engine/test/configuration/applier/xrddefault.cc:67]: No log_v2?

File: centreon-engine/test/configuration/applier/xrddefault.cc [hunk 26]
```diff
@@ -157,7 +158,7 @@ int xrddefault_initialize_retention_data(char const* config_file) {
     if ((xrddefault_retention_file_fd =
              open(xrddefault_retention_file, O_WRONLY | O_CREAT,
                   S_IRUSR | S_IWUSR)) == -1) {
-      logger(log_runtime_error, basic)
+      engine_logger(log_runtime_error, basic)
```
Comment (bouda1) [inline centreon-engine/test/configuration/applier/xrddefault.cc:161]: No log_v2?

File: centreon-engine/test/configuration/applier/xrddefault.cc [hunk 27]
```diff
@@ -189,11 +190,11 @@ int xrddefault_cleanup_retention_data(char const* config_file) {
 /******************************************************************/
 
 int xrddefault_save_state_information() {
-  logger(dbg_functions, basic) << "xrddefault_save_state_information()";
+  engine_logger(dbg_functions, basic) << "xrddefault_save_state_information()";
```
Comment (bouda1) [inline centreon-engine/test/configuration/applier/xrddefault.cc:193]: no log_v2?

File: centreon-engine/test/configuration/applier/xrddefault.cc [hunk 28]
```diff
@@ -189,11 +190,11 @@ int xrddefault_cleanup_retention_data(char const* config_file) {
...
 
   /* make sure we have everything */
   if (xrddefault_retention_file == NULL || xrddefault_retention_file_fd == -1) {
-    logger(log_runtime_error, basic)
+    engine_logger(log_runtime_error, basic)
```
Comment (bouda1) [inline centreon-engine/test/configuration/applier/xrddefault.cc:197]: same

File: centreon-engine/test/configuration/applier/xrddefault.cc [hunk 29]
```diff
@@ -613,7 +614,7 @@ int xrddefault_save_state_information() {
       (fsync(xrddefault_retention_file_fd) == -1) ||
       (lseek(xrddefault_retention_file_fd, 0, SEEK_SET) == (off_t)-1)) {
     char const* msg(strerror(errno));
-    logger(log_runtime_error, basic)
+    engine_logger(log_runtime_error, basic)
```
Comment (bouda1) [inline centreon-engine/test/configuration/applier/xrddefault.cc:617]: same...

File: centreon-engine/test/configuration/applier/xrddefault.cc [hunk 30]
```diff
@@ -697,11 +698,11 @@ int xrddefault_read_state_information() {
   double runtime[2];
   int found_directive = FALSE;
 
-  logger(dbg_functions, basic) << "xrddefault_read_state_information()";
+  engine_logger(dbg_functions, basic) << "xrddefault_read_state_information()";
 
   /* make sure we have what we need */
   if (xrddefault_retention_file == NULL) {
-    logger(log_runtime_error, basic)
+    engine_logger(log_runtime_error, basic)
```
Comment (bouda1) [inline centreon-engine/test/configuration/applier/xrddefault.cc:705]: same

File: centreon-engine/test/configuration/applier/xodtemplate.cc [hunk 31]
```diff
@@ -552,7 +553,7 @@ int xodtemplate_process_config_dir(char* dirname, int options) {
   /* open the directory for reading */
   dirp = opendir(dirname);
   if (dirp == NULL) {
-    logger(log_config_error, basic)
+    engine_logger(log_config_error, basic)
```
Comment (bouda1) [inline centreon-engine/test/configuration/applier/xodtemplate.cc:556]: no log_v2 in this file?

File: centreon-engine/test/configuration/applier/events.cc [hunk 32]
```diff
@@ -686,6 +769,7 @@ void init_timing_loop() {
     printf("\n\n");
   }
 
-  logger(dbg_functions, basic) << "init_timing_loop()";
+  engine_logger(dbg_functions, basic) << "init_timing_loop()";
+  log_v2::functions()->trace "init_timing_loop()"();
```
Comment (bouda1) [inline centreon-engine/test/configuration/applier/events.cc:773]: How this could compile?

File: centreon-engine/src/service.cc [hunk 33]
```diff
@@ -1009,6 +1043,17 @@ int service::handle_async_check_result(check_result* queued_check_result) {
...
       << ", OUTPUT: " << queued_check_result->get_output();
+  log_v2::checks()->debug(
+      "HOST: {}, SERVICE: {}, CHECK TYPE: {}, OPTIONS: {}, RESCHEDULE: {}, "
+      "EXITED OK: {}, EXEC TIME: {}, return CODE: {}, OUTPUT: {}",
+      _hostname, _description,
+      (queued_check_result->get_check_type() == check_active ? "Active"
+                                                             : "Passive"),
+      queued_check_result->get_check_options(),
+      (queued_check_result->get_reschedule_check() ? "Yes" : "No"),
+      (queued_check_result->get_exited_ok() ? "Yes" : "No"), execution_time,
```
Comment (bouda1) [inline centreon-engine/src/service.cc:1054]: too many parenthesis

File: centreon-engine/src/service.cc [hunk 34]
```diff
@@ -1167,6 +1234,12 @@ int service::handle_async_check_result(check_result* queued_check_result) {
         << "\n"
         << "Perf Data:\n"
         << (get_perf_data().empty() ? "NULL" : get_perf_data());
+    log_v2::checks()->debug(
+        "Parsing check output...\n"
+        "Short Output:\n {} \n Long Output:\n {} \n Perf Data:\n {}",
+        (get_plugin_output().empty() ? "NULL" : get_plugin_output()),
+        (get_long_plugin_output().empty() ? "NULL" : get_long_plugin_output()),
+        (get_perf_data().empty() ? "NULL" : get_perf_data()));
```
Comment (bouda1) [inline centreon-engine/src/service.cc:1242]: \n

File: centreon-engine/src/service.cc [hunk 35]
```diff
@@ -2010,10 +2144,13 @@ void service::check_for_flapping(bool update,
     if ((_current_state != service::state_ok) || get_is_flapping())
       is_flapping = true;
   }
-  logger(dbg_flapping, more)
+  engine_logger(dbg_flapping, more)
       << com::centreon::logging::setprecision(2) << "Service "
       << (is_flapping ? "is" : "is not") << " flapping ("
       << curved_percent_change << "% state change).";
+  log_v2::checks()->debug("Service {} flapping ({}% state change).",
+                          (is_flapping ? "is" : "is not"),
```
Comment (bouda1) [inline centreon-engine/src/service.cc:2152]: parenthesis

File: centreon-engine/src/service.cc [hunk 36]
```diff
@@ -2416,18 +2614,25 @@ int service::run_async_check(int check_options,
...
 
-  logger(dbg_checks, basic)
+  engine_logger(dbg_checks, basic)
       << "Scheduling a "
       << (options & CHECK_OPTION_FORCE_EXECUTION ? "forced" : "non-forced")
       << ", active check of service '" << _description << "' on host '"
       << _hostname << "' @ " << my_ctime(&check_time);
+  log_v2::checks()->trace(
+      "Scheduling a {}, active check of service '{}' on host '{}' @ {}",
+      (options & CHECK_OPTION_FORCE_EXECUTION ? "forced" : "non-forced"),
```
Comment (bouda1) [inline centreon-engine/src/service.cc:2627]: ()

**Review by bouda1 — CHANGES_REQUESTED** [review]

File: centreon-engine/test/configuration/applier/xodtemplate.cc (hunk 31 above)
Comment (rem31) [reply centreon-engine/test/configuration/applier/xodtemplate.cc:556]: is not compiled

File: centreon-engine/test/configuration/applier/xrddefault.cc (hunk 29 above)
Comment (rem31) [reply centreon-engine/test/configuration/applier/xrddefault.cc:617]: is not compiled

File: centreon-engine/test/configuration/applier/xrddefault.cc (hunk 30 above)
Comment (rem31) [reply centreon-engine/test/configuration/applier/xrddefault.cc:705]: is not compiled

File: centreon-engine/src/main.cc [hunk 37]
```diff
@@ -173,35 +174,38 @@ int main(int argc, char* argv[]) {
...
-          << "<http://www.gnu.org/licenses/>.";
+             "along with this program. If not, see\n"
+             "<http://www.gnu.org/licenses/>.\n";
+
       retval = EXIT_SUCCESS;
     }
     // If requested or if an error occured, print usage.
     else if (error || display_help) {
-      logger(logging::log_info_message, logging::basic)
+      std::cout
```
Comment (bouda1) [inline centreon-engine/src/main.cc:208]: Could you also remove the "<<" here pleaaaaaaaase?

File: centreon-engine/src/main.cc [hunk 38]
```diff
@@ -244,39 +250,51 @@ int main(int argc, char* argv[]) {
...
+                  << "Checked " << serviceescalation::serviceescalations.size()
+                  << " service escalations.\n"
+                  << "Checked " << servicegroup::servicegroups.size()
+                  << " service groups.\n"
+                  << "Checked " << service::services.size() << " services.\n"
+                  << "Checked " << timeperiod::timeperiods.size()
+                  << " time periods.\n"
+                  << "\n"
+                  << "Total Warnings: " << config_warnings << "\n"
+                  << "Total Errors:   " << config_errors << std::endl;
```
Comment (bouda1) [inline centreon-engine/src/main.cc:277]: Here too, you can remove several "<<"

File: centreon-engine/src/main.cc [hunk 39]
```diff
@@ -296,7 +314,10 @@ int main(int argc, char* argv[]) {
           try {
             p.parse(config.state_retention_file(), state);
           } catch (std::exception const& e) {
-            logger(logging::log_config_error, logging::basic) << e.what();
+            std::cout << "Error while parsing the retention: {}" << e.what()
+                      << std::endl;
+            engine_logger(logging::log_config_error, logging::basic)
+                << e.what();
```
Comment (bouda1) [inline centreon-engine/src/main.cc:320]: is it really needed?

File: centreon-engine/src/main.cc [hunk 40]
```diff
@@ -296,7 +314,10 @@ int main(int argc, char* argv[]) {
           try {
             p.parse(config.state_retention_file(), state);
           } catch (std::exception const& e) {
-            logger(logging::log_config_error, logging::basic) << e.what();
+            std::cout << "Error while parsing the retention: {}" << e.what()
+                      << std::endl;
```
Comment (bouda1) [inline centreon-engine/src/main.cc:318]: should be a log_v2...

File: centreon-engine/src/main.cc [hunk 41]
```diff
@@ -306,7 +327,7 @@ int main(int argc, char* argv[]) {
         display_scheduling_info();
         retval = EXIT_SUCCESS;
       } catch (std::exception const& e) {
-        logger(logging::log_config_error, logging::basic) << e.what();
+        std::cout << e.what() << std::endl;
```
Comment (bouda1) [inline centreon-engine/src/main.cc:330]: to check...

**Review by bouda1 — CHANGES_REQUESTED** [review]

File: centreon-engine/modules/external_commands/src/processing.cc [hunk 42]
```diff
@@ -639,16 +643,25 @@ bool processing::execute(const std::string& cmdstr) const {
...
-  logger(dbg_external_command, more) << "External command id: " << command_id
-                                     << "\nCommand entry time: " << entry_time
-                                     << "\nCommand arguments: " << args;
+  engine_logger(dbg_external_command, more)
+      << "External command id: " << command_id
+      << "\nCommand entry time: " << entry_time
+      << "\nCommand arguments: " << args;
+  log_v2::external_command()->debug("External command id: {}", command_id);
+  log_v2::external_command()->debug("Command entry time: {}", entry_time);
+  log_v2::external_command()->debug("Command arguments: {}", args);
```
[the remark this answers is not in the document]
Comment (rem31) [reply centreon-engine/modules/external_commands/src/processing.cc:664]: is a debug log

File: centreon-engine/src/commands/connector.cc [hunk 43]
```diff
@@ -396,8 +434,10 @@ void connector::_connector_close() {
   if (is_timeout || !_query_quit_ok) {
     _process.kill();
     if (is_timeout)
-      logger(log_runtime_warning, basic)
+      engine_logger(log_runtime_warning, basic)
           << "Warning: Cannot close connector '" << _name << "': Timeout";
+    log_v2::runtime()->warn("Warning: Cannot close connector '{}': Timeout",
+                            _name);
```
Comment (bouda1) [inline centreon-engine/src/commands/connector.cc:440]: missing {}

**Review by bouda1 — APPROVED** [review]

**Pull request merged**